src/openforms/forms/admin/views.py

```diff
@@ -166,3 +168,92 @@ def form_fields():
+                " Please verify the selected forms have the ogone-legacy payment backend"
+                " configured."
+            ),
+        },
+        label=_("Forms"),
+    )
+
+    outdated_forms: list[Form] = []
+    merchant_mapping: dict[int, OgoneMerchant] = {}
+
+    def clean(self) -> dict:
+        cleaned_data = super().clean()
+        forms_to_migrate = cleaned_data.get("forms_to_migrate", [])
+
+        merchants: dict[int, Form] = {
+            form.payment_backend_options["merchant_id"]: form
+            for form in forms_to_migrate
+            if "merchant_id" in form.payment_backend_options
+        }
+
+        found_merchants: dict[int, OgoneMerchant] = {
+            merchant.pk: merchant
+            for merchant in OgoneMerchant.objects.filter(id__in=list(merchants.keys()))
+        }
+
+        for merchant_id in merchants:
+            if merchant_id not in found_merchants:
+                self.outdated_forms.append(merchants[merchant_id])
+
+        self.merchant_mapping = {
+            form.id: found_merchants[form.payment_backend_options["merchant_id"]]
+            for form in forms_to_migrate
+            if form.payment_backend_options["merchant_id"] in found_merchants
+        }
+
+        return cleaned_data
+
+    def save(self, **kwargs) -> None:
+        with transaction.atomic():
+            for form in self.cleaned_data["forms_to_migrate"]:
+                if not (ogone_merchant := self.merchant_mapping.get(form.id)):
+                    continue
+
+                form.payment_backend = "worldline"
+                if "com_template" in form.payment_backend_options:
+                    form.payment_backend_options = {
+                        "merchant": ogone_merchant.pspid,
+                        "descriptor_template": form.payment_backend_options[
+                            "com_template"
+                        ],
+                    }
+                else:
+                    form.payment_backend_options = {"merchant": ogone_merchant.pspid}
+
+                form.save()
+
+
+class PaymentMigrationView(PermissionRequiredMixin, FormView):
+    form_class = PaymentMigrationForm
+    template_name = "admin/forms/form/migrate-payment-backend.html"
+    permission_required = "forms.change_form"
+    success_url = reverse_lazy("admin:forms_form_changelist")
+
+    def form_valid(self, form):
+        form.save()
+
+        if form.outdated_forms:
+            message = _(
+                "The following forms reference a merchant that is non-existent: %s"
+                "Please select a new merchant for these forms."
+            ) % ",".join(str(form.name) for form in form.outdated_forms)
+
+            message_level = messages.WARNING
+        else:
+            message = _("%s forms were migrated.") % len(
+                form.cleaned_data["forms_to_migrate"]
+            )
+            message_level = messages.SUCCESS
```

Comment:
this is weird - the branching logic here implies one or the other happens, but both can happen at the same time. Part of the forms can be outdated, part of the forms can be migrated.